vita3k/codec/include/codec/state.h

```diff
@@ -121,14 +122,28 @@ struct H264DecoderState : public DecoderState {
     ~H264DecoderState() override;
 };
 
+struct MJpegPitch {
+    uint32_t x;
+    uint32_t y;
+};
+
+struct MJpegDecoderOptions {
+    bool use_standard_decoder;
+    int downscale_ratio;
+};
+
 struct MjpegDecoderState : public DecoderState {
-    uint32_t mcu_width;
-    uint32_t mcu_height;
+    bool use_standard_decoder = false;
+    int downscale_ratio = 1;
+
+    MJpegPitch pitch[4];
```

Comment:
Is there a reason for pitch to be an array of size 4? All the formats considered have at most 3 planes (I know the pitch field of SceJpegOutputInfo has size 4, but afaik the last one is never used).

Reply:
Theoretically, the YCCK type of JPEG format can have 4 color planes, and the PS Vita header seems to be designed with this situation in mind. Also, the PS Vita's default gallery app supports YCCK type JPEGs.
However, SceJpeg doesn't support YCCK type images. So I can't know the implementation details of how the PS Vita's gallery app is implemented. For example, it's possible that SceJpegArm supports YCCK type images. Or Sony might have implemented a dedicated decoder in their own apps.
The reason for setting the pitch to 4 in the current implementation is because there might be software on the PS Vita that uses 4 planes, and I didn't have enough confidence to reduce it to 3 at the cost of losing consistency with the PS Vita. It would be possible to implement it with 3 first, and then expand it if situations using all 4 are discovered later. Which approach do you think is better?

Reply:
Well, I think it's better to write the code assuming only at most 3 planes, but the way it is right now is fine.